tests/modules/nf-core/paragraph/idxdepth/main.nf

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { PARAGRAPH_IDXDEPTH } from '../../../../../modules/nf-core/paragraph/idxdepth/main.nf'
+
+workflow test_paragraph_idxdepth_bam {
+
+    input = [
+        [ id:'test', single_end:false ], // meta map
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true)
+    ]
+
+    fasta = [
+        [ id:'fasta' ], // meta map
+        file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+    ]
+
+    fasta_fai = [
+        [ id:'fasta_fai' ], // meta map
+        file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+    ]
+
+    PARAGRAPH_IDXDEPTH (
+        input,
+        fasta,
+        fasta_fai
+    )
+}
+
+workflow test_paragraph_idxdepth_cram {
+
+    input = [
+        [ id:'test', single_end:false ], // meta map
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true)
+    ]
+
+    fasta = [
+        [ id:'fasta' ], // meta map
+        file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+    ]
+
+    fasta_fai = [
+        [ id:'fasta_fai' ], // meta map
+        file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+    ]
```

Comment:
That was a discussion we had a while back with the maintainers group. Basically it's advised to add meta to each input tuple to be able to add meta for all input files instead of only the top ones

Reply:
OK, just looks weird when you have to construct a meta object like that. Would it be better to reuse the meta object in all channels? (Maybe that's what you do in the actual pipeline.)

Reply:
Not really, some pipelines want to have different meta for their fasta references (e.g. species, assembly...). I don't need it but it's easier to add an empty meta map than to patch the module IMO

Reply:
Fine.